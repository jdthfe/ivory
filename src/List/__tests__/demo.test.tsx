import webDemoTest from '@tests/shared/demoTest';
webDemoTest('List');
