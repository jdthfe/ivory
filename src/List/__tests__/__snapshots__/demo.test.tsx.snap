// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List correctly 1`] = `
<div>
  <div
    class="List"
  >
    <div
      class="eui-wingblank eui-wingblank-l"
    >
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Basic Style
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-thumb"
            >
              <img
                src="https://avatars1.githubusercontent.com/u/49478572?s=200&v=4"
              />
            </div>
            <div
              class="eui-list-content"
            >
              Title
            </div>
            <div
              class="eui-list-extra"
            >
              extra content
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Subtitle
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content eui-list-content-wrap"
            >
              Title 
              <div
                class="eui-list-brief"
              >
                subtitle
              </div>
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-thumb"
            >
              <img
                src="https://avatars1.githubusercontent.com/u/49478572?s=200&v=4"
              />
            </div>
            <div
              class="eui-list-content"
            >
              Title 
              <div
                class="eui-list-brief"
              >
                subtitle
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Customized Right Side（Empty Content / Text / Image）
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              Title
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              Title
            </div>
            <div
              class="eui-list-arrow eui-list-arrow-horizontal"
            >
              <svg
                aria-hidden="true"
                class="eui-icon eui-icon-l"
              >
                <use
                  xlink:href="#arrow"
                />
              </svg>
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              Title
            </div>
            <div
              class="eui-list-extra"
            >
              extra content
            </div>
            <div
              class="eui-list-arrow eui-list-arrow-horizontal"
            >
              <svg
                aria-hidden="true"
                class="eui-icon eui-icon-l"
              >
                <use
                  xlink:href="#arrow"
                />
              </svg>
            </div>
          </div>
          <div
            class="eui-list-item eui-list-align-top"
          >
            <div
              class="eui-list-thumb"
            >
              <img
                src="https://avatars1.githubusercontent.com/u/49478572?s=200&v=4"
              />
            </div>
            <div
              class="eui-list-content"
            >
              Title 
              <div
                class="eui-list-brief"
              >
                subtitle
              </div>
            </div>
            <div
              class="eui-list-extra"
            >
              10:30
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Align Vertical Center
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              Title 
              <div
                class="eui-list-brief"
              >
                subtitle
              </div>
            </div>
            <div
              class="eui-list-extra"
            >
              extra content
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list"
      >
        <div
          class="eui-list-header"
        >
          Icon in the left
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-thumb"
            >
              <img
                src="https://avatars1.githubusercontent.com/u/49478572?s=200&v=4"
              />
            </div>
            <div
              class="eui-list-content"
            >
              My wallet
            </div>
            <div
              class="eui-list-arrow eui-list-arrow-horizontal"
            >
              <svg
                aria-hidden="true"
                class="eui-icon eui-icon-l"
              >
                <use
                  xlink:href="#arrow"
                />
              </svg>
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-thumb"
            >
              <img
                src="https://avatars1.githubusercontent.com/u/49478572?s=200&v=4"
              />
            </div>
            <div
              class="eui-list-content"
            >
              My Cost Ratio
            </div>
            <div
              class="eui-list-arrow eui-list-arrow-horizontal"
            >
              <svg
                aria-hidden="true"
                class="eui-icon eui-icon-l"
              >
                <use
                  xlink:href="#arrow"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Text Wrapping
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
            data-seed="logId"
          >
            <div
              class="eui-list-content"
            >
              Single line，long text will be hidden with ellipsis；
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content eui-list-content-wrap"
            >
              Multiple line，long text will wrap；Long Text Long Text Long Text Long Text Long Text Long Text
            </div>
          </div>
          <div
            class="eui-list-item eui-list-align-top"
          >
            <div
              class="eui-list-content eui-list-content-wrap"
            >
              Multiple line and long text will wrap. Long Text Long Text Long Text
            </div>
            <div
              class="eui-list-extra"
            >
              extra content
            </div>
          </div>
          <div
            class="eui-list-item spe"
          >
            <div
              class="eui-list-content"
            >
              In rare cases, the text of right side will wrap in the single line with long text. long text long text long text
            </div>
            <div
              class="eui-list-extra"
            >
              no arrow
            </div>
            <div
              class="eui-list-arrow eui-list-arrow-empty"
            >
              <svg
                aria-hidden="true"
                class="eui-icon eui-icon-l"
              >
                <use
                  xlink:href="#arrow"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list my-list"
      >
        <div
          class="eui-list-header"
        >
          Other
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              Click to disable
            </div>
          </div>
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              <select>
                <option
                  selected=""
                  value="1"
                >
                  Html select element
                </option>
                <option
                  disabled=""
                  value="2"
                >
                  Unable to select
                </option>
                <option
                  value="3"
                >
                  option 3
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="eui-list"
      >
        <div
          class="eui-list-header"
        >
          Form Validation
        </div>
        <div
          class="eui-list-body"
        >
          <div
            class="eui-list-item"
          >
            <div
              class="eui-list-content"
            >
              <button
                class="eui-button eui-button-l eui-button-primary eui-button-inline"
                role="button"
              >
                Submit
              </button>
              <button
                class="eui-button eui-button-l eui-button-primary eui-button-inline"
                role="button"
                style="margin-left: 2.5px;"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
